gapic/src/main/com/google/gapid/models/Profile.java

```diff
@@ -75,7 +78,7 @@ protected boolean shouldLoad(Capture.Data data) {
 
   @Override
   protected ListenableFuture<Data> doLoad(Source source, Path.Device device) {
-    return transform(client.profile(capture.getData().path, device), r -> new Data(device, r));
+    return transform(client.profile(capture.getData().path, device, this.loopCount), r -> new Data(device, r));
```

Comment:
For the issue mentioned in the PR description, about "the button toggle change will only take effect after restart", it is related with this part. The setting preference is actually saved when the dialog closes, but the Profile class here fixes loopCount variable at the initialization time based on the setting condition at that time. 
Instead we could re-query the settings here and get the most updated preference to do the server query.
```suggestion
    int loopCount = settings.preferences().getUseFrameLooping() ? FRAME_LOOP_COUNT : <?>;
    return transform(client.profile(capture.getData().path, device, loopCount), r -> new Data(device, r));
```

Reply:
Really good comment. Thanks for spotting where the problem was happening!